### PR TITLE
helm install with "CCM_COUNT" number of ccm Pods

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

 

 
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
helm install with "CCM_COUNT" number of ccm Pods
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
helm install with "CCM_COUNT" number of ccm Pods
```

### CODE REVIEW

**Comment by lzhecheng** [comment]
/hold
waiting for https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1790

**Comment by jackfrancis** [comment]
Is this a new best practice? To have as many cloud-controller-manager pods as there are nodes? Seems like using either 1 or some static number that fulfills redundancy requirements (e.g., 3) would make more sense.

**Comment by lzhecheng** [comment]
> Is this a new best practice? To have as many cloud-controller-manager pods as there are nodes? Seems like using either 1 or some static number that fulfills redundancy requirements (e.g., 3) would make more sense.

Yes, maybe not be the same as control plane node number but provide an environment variable instead?

**Comment by lzhecheng** [comment]
@jackfrancis PTAL. Thanks.

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Review by jackfrancis — APPROVED** [review]
/lgtm

File: docs/book/src/developers/development.md [hunk 1]
```diff
@@ -559,6 +559,7 @@ You can optionally set the following variables:
 | `TEST_WINDOWS`                  | Build a cluster that has Windows worker nodes.          |
 | `REGISTRY`                  | Registry to push any custom k8s images or cloud provider images built.                                                                     |
 | `CLUSTER_TEMPLATE`          | Use a custom cluster template. It can be a path to a template under templates/, a path on the host or a link. If the value is not set, the script will choose the appropriate cluster template based on existing environment variables. |
+| `CCM_COUNT` | Set the number of cloud-controller-manager, which should not be more than control plane Node number. |
```
[the remark this answers is not in the document]
Comment (lzhecheng) [reply docs/book/src/developers/development.md]: Updated.

**Pull request merged**